CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/parameters/SelectOptionsExtension.java

```diff
@@ -0,0 +1,95 @@
+/*
+* Copyright 2010-2023 Australian Signals Directorate
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+package au.gov.asd.tac.constellation.plugins.parameters;
+
+import javafx.scene.control.MenuButton;
+import javafx.scene.control.MenuItem;
+import javafx.scene.control.Tooltip;
+import javafx.event.ActionEvent;
+import javafx.event.EventHandler;
+import javafx.scene.control.ContextMenu;
+import org.controlsfx.control.CheckComboBox;
+
+/**
+ * This component enables selection options for input fields extending CheckComboBox's
+ * By Default it enables Select All and Clear All functionality but also permits custom menu selection options. 
+ * Selection options can be displayed in a Menu Button or in a context menu accessed through right clicking the input field. 
+ * 
+ * @author capricornunicorn123
+ */
+public final class SelectOptionsExtension{
+    
+    private final CheckComboBox PARENT_INPUT_FIELD;
+    private final ContextMenu CONTEXT_MENU = new ContextMenu();
+    private final MenuButton MENU_BUTTON = new MenuButton("...");
```

Comment:
Coding convention ... all caps should only be used for fixed static variables.
It looks like all these variables are tied to separate instances of `SelectOptionsExtension`, so they should be renamed in camelCase format.